pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -360,6 +360,14 @@ def _check_deprecated_callback_hooks(trainer: "pl.Trainer") -> None:
                 "The `on_before_accelerator_backend_setup` callback hook was deprecated in"
                 " v1.6 and will be removed in v1.8. Use `setup()` instead."
             )
+        if is_overridden(method_name="on_load_checkpoint", instance=callback):
+            rank_zero_deprecation(
```

Comment:
this should be a warning but not necessarily a deprecation. otherwise someone who's overriding this method to do some processing on the fully loaded checkpoint will still see this message.

Reply:
they should see the message though? they expect `callback_state` but it'll become entire `checkpoint` dict

Reply:
Ok, I see what you mean. Deprecation makes sense then